src/plugins/data/public/search/search_service.ts

```diff
@@ -268,9 +266,30 @@ export class SearchService implements Plugin<ISearchSetup, ISearchStart> {
     return {
       aggs,
       search,
-      showError: (e: Error) => {
+      showError: (e) => {
         this.searchInterceptor.showError(e);
       },
+      showWarnings: (inspector, cb) => {
+        inspector.adapter?.getRequests().forEach((req) => {
+          let handled: boolean | undefined;
```

Comment:
what about something in this direction

```
const warnings = extractWarnings(...).filter(warning => {
  return cb && cb(warning);
});
if (warnings.length) handleResponse(warnings);

```

Reply:
I am not super in favor of arranging it this way. I think `handleResponse` should still have access to the entire response. Receiving only the warnings make it not have the response at all... so it should be named `handleWarnings`. 

On the other hand, this PR doesn't require major changes to inspector and handleResponse, right? I think the gist of your feedback is, allow the consumer to give a yes/no per each warning on whether we need to prevent the default toast notification.

Reply:
i am definitely ok with renaming that method. its not handling the response now either.
yes, user needs to have this choice for every warning, not all or nothing. I think going forward without callbacks might be easier (as suggested above)